overcast/runner/__init__.py

```diff
@@ -128,6 +128,83 @@ def get_creds_from_env():
 #    d['cacert'] = os.environ.get('OS_CACERT', None)
     return d
 
+
+class node(object):
+    dr = self.DeploymentRunner()
+
+    def __init__(self, name, info, keypair, userdata, mappings=None, record_resource=None):
+        self.record_resource = lambda *args, **kwargs: None
+        self.info     = info
+        self.networks = dr.networks or {}
+        self.secgroups = dr.secgroups or {}
+        self.keypair  = keypair
+        self.userdata = userdata
+        self.image    = None
+        self.flavor   = None
+        self.mappings = mappings or {}
+        self.server_id = None
+        self.server_status = None
+
+    def poll(self, desired_status = 'ACTIVE'):
+        if self.server_status != desired_status:
+            nc = dr.get_nova_client()
+            self.server_status = nc.servers.get(self.server_id).status
+        return self.server_status
+
+    def clean(self):
+        nc = dr.get_nova_client()
+        server = nc.servers.get(self.server_id)
+        server.delete()
+
+    def build(self):
+        nc = dr.get_nova_client()
+        if info['image'] in self.mappings.get('images', {}):
+            info['image'] = self.mappings['images'][info['image']]
+
+        if info['flavor'] in self.mappings.get('flavors', {}):
+            info['flavor'] = self.mappings['flavors'][info['flavor']]
+
+        image = nc.images.get(info['image'])
+        flavor = nc.flavors.get(info['flavor'])
```

Comment:
You could do this in __init__ to avoid doing it more than once (in case of a rebuild).